app/locations/controllers.js

```diff
@@ -1,10 +1,25 @@
-const { sortBy } = require('lodash')
+const { get, sortBy } = require('lodash')
 
-function locations(req, res) {
-  const locations = sortBy(req.userLocations, 'title')
+const referenceDataService = require('../../common/services/reference-data')
+
+async function locations(req, res, next) {
+  const userPermissions = get(req.session, 'user.permissions', [])
+
+  let locations = []
+  let regions = []
+
+  if (userPermissions.includes('allocation:create')) {
+    try {
+      regions = await referenceDataService.getRegions()
+    } catch (error) {
+      return next(error)
+    }
+  }
+  locations = sortBy(req.userLocations, 'title')
```

Comment:
Any reason to not set this on line 9 (or even 7 next to the other `const` statement) and use `const`? Doesn't appear to be reset anyway so could combine that statement.